src/net_processing.cpp

```diff
@@ -775,6 +775,7 @@ static CCriticalSection cs_most_recent_block;
 static std::shared_ptr<const CBlock> most_recent_block;
 static std::shared_ptr<const CBlockHeaderAndShortTxIDs> most_recent_compact_block;
 static uint256 most_recent_block_hash;
+static bool fWitnessesPresentInMostRecentCompactBlock;
```

Comment:
nit: A comment above saying that all these fields are protected by cs_most_recent_block would be nice since it's a growing list.